src/PrestaShopBundle/Translation/Provider/ExternalModuleLegacySystemProvider.php

```diff
@@ -94,43 +110,37 @@ public function getIdentifier()
      */
     public function setModuleName($moduleName)
     {
-        if (
-            null === $this->moduleName || empty($this->moduleName)
-        ) {
+        if (null === $this->moduleName || empty($this->moduleName)) {
             UnsupportedModuleException::moduleNotProvided(self::getIdentifier());
         }
 
         $this->moduleName = $moduleName;
 
+        // ugly hack, I know
+        $this->domain = DomainHelper::buildModuleBaseDomain($moduleName);
+
         return $this;
     }
 
+    /**
+     * @param string $domain
+     *
+     * @return AbstractProvider|SearchProviderInterface|void
+     */
+    public function setDomain($domain)
```

Comment:
Nope because the domain name is inferred from translations themselves. There's no way that you could define it from the outside.

The providers structure is not well designed, their getters are not pure functions as the output depends on what setters you called first. Unfortunately this cannot be fixed without rewriting the whole system.